### PR TITLE
feat: adding support for --component flag

### PR DESCRIPTION
**Pull request by skiano**
This is an attempt to resolve https://github.com/egoist/poi/issues/243

I tried to just follow what you outlined in the issue. Let me know what you think 😄

### CODE REVIEW

File: packages/poi/test/config.test.js [hunk 1]
```diff
@@ -197,4 +201,21 @@ describe('get webpack config', () => {
       expect(config.entry.foo).toEqual(['foo', 'bar'])
     })
   })
+
+  describe('build component', () => {
+    it('respects component option', async () => {
```
Comment (egoist) [inline packages/poi/test/config.test.js]: I think we can just require `html-webpack-plugin` here?

File: packages/poi/lib/handle-options.js [hunk 2]
```diff
@@ -9,6 +9,14 @@ const { inferHTML, readPkg } = require('./utils')
 module.exports = co.wrap(function * (options) {
   const loadExternalConfig = new LoadExternalConfig({ cwd: options.cwd })
 
+  if (options.component) {
+    console.log('> Bundling component')
+    options.format = 'cjs'
```
Comment (egoist) [inline packages/poi/lib/handle-options.js]: what about supporting `umd` format as well, like:

```bash
poi --component {LibraryName}
```

When `options.component` is a string we build it in UMD format.

Comment (skiano) [reply packages/poi/lib/handle-options.js]: Ok, sounds good

File: packages/poi/test/config.test.js (hunk 1 above)
Comment (skiano) [reply packages/poi/test/config.test.js]: that would probably be clearer, yes

**Comment by skiano** [comment]
I added support for `poi --component {LibraryName}` and added an error if you pass `umd` but no module name and an error if you pass anything but `umd` or `cjs`

**Comment by egoist** [comment]
The `LibraryName` here actually means `moduleName` that we use in `umd` format 😅 

So `--component foo` should be taken as `--format umd --module-name foo`

**Comment by skiano** [comment]
Oh I see! So no name means use `cjs`, but if you put a name, it implies umd. I can make those changes

**Comment by skiano** [comment]
@egoist did those changes look how you expect?

File: packages/poi/package.json [hunk 3]
```diff
@@ -19,6 +19,7 @@
   "bin": "bin/cli.js",
   "scripts": {
     "test": "jest",
+    "test:watch": "jest --watch",
```
Comment (egoist) [inline packages/poi/package.json]: `yarn test --watch` or use `npx` will do the trick 😄

**Comment by egoist** [comment]
Hmm I'm not sure if we should set `minimize: false` by default for *component*

**Comment by skiano** [comment]
@egoist perhaps, but for umd wouldn't you want it `true` by default? I guess i was thinking it is easy enough to set it to `false` yourself.

Also if someone doesn't think about it, isn't it safer for it to be `true`

**Comment by skiano** [comment]
@egoist I totally agree. I think I just did that based on the issue conversation. But wouldnt mind for it to be true by default.

**Pull request merged**